### PR TITLE
Fixed duplicate variant bug.

### PR DESCRIPTION
**Pull request by cseed**
Improved BGzipCodecSuite to reproduce bug.
Added hidden BGZipBlocks command.

### CODE REVIEW

**Comment by tpoterba** [comment]
Fixes #869

File: src/main/scala/org/broadinstitute/hail/check/Gen.scala [hunk 1]
```diff
@@ -208,27 +208,27 @@ object Gen {
...
-        val s = p.rng.nextInt(min, p.size)
-        val part = partition(p.rng, p.size, s)
-        val t = mutable.Set.empty[T]
-        for (i <- 0 until s) {
-          var element = g.resize(part(i))(p)
-          while (t.contains(element)) {
-            element = g.resize(part(i))(p)
-          }
-          t += element
+  Gen { (p: Parameters) =>
```
Comment (tpoterba) [inline src/main/scala/org/broadinstitute/hail/check/Gen.scala]: unnecessary formatting change

File: src/test/scala/org/broadinstitute/hail/io/compress/BGzipCodecSuite.scala [hunk 2]
```diff
@@ -1,69 +1,118 @@
...
+    val g = for (n <- Gen.oneOfGen(
+      Gen.choose(0, 10),
+      Gen.choose(0, 100));
+    splits <- Gen.buildableOfN[Array, Long](n,
+      Gen.oneOfGen(Gen.choose(0, compLength),
+        Gen.applyGen(Gen.oneOf[(Long) => Long](identity, _ - 1, _ + 1),
+          Gen.oneOfSeq(compSplits))))
+      .map { splits =>
+        (Array(0L, compLength) ++ splits).distinct.sorted
+      }) yield splits
```
Comment (danking) [inline src/test/scala/org/broadinstitute/hail/io/compress/BGzipCodecSuite.scala]: nitpick: I find this hard to read, though I'm not sure there's much to be done given Scala's syntax. Maybe this is a bit better?

```scala
    val g = for (n <- Gen.oneOfGen(
      Gen.choose(0, 10),
      Gen.choose(0, 100));
      validSplit = Gen.oneOfGen[Long](Gen.choose(0, compLength),
        Gen.applyGen(Gen.oneOf[(Long) => Long](identity, _ - 1, _ + 1), 
          Gen.oneOfSeq(compSplits)));
      splits <- Gen.buildableOfN[Array, Long](n, validSplit))
      yield (Array(0L, compLength) ++ splits).distinct.sorted
```

File: src/main/scala/org/broadinstitute/hail/check/Gen.scala (hunk 1 above)
Comment (cseed) [inline src/main/scala/org/broadinstitute/hail/check/Gen.scala]: My IntelliJ is doing this and I can't figure out why on this function in particular.  Thoughts?

Comment (tpoterba) [inline src/main/scala/org/broadinstitute/hail/check/Gen.scala]: I'm happy to just merge (this is only formatting) but it looks like the `Gen {` has the same indentation as the function declaration above it, which seems wrong.  Maybe try rebooting intellij and see if that fixes it?  Otherwise let's put this in and rebuild

**Comment by cseed** [comment]
Addressed comments.  Rebooting IntelliJ didn't fix it.  I added some extra braces and now it looks correct.  Also included @danking's suggestion.
